regenerate.php

```diff
@@ -40,11 +40,12 @@
 $error = '';
 
 if ($fromform = $mform->get_data()) {
-    $error = auth_saml2_process_regenerate_form($fromform);
-    if ($error === '') {
-        redirect("$CFG->wwwroot/admin/settings.php?section=authsettingsaml2");
+    // Check if the certificates are locked, in that case don't regenerate.
```

Comment:
I think this checking logic isn't in the best place, I think create_certificate is where the most important check should be, and if it fails it throws a hard exception. Then its up to whatever that calls it to handle it, or not, and avoid it in advance



auth_saml2_process_regenerate_form

Reply:
@brendanheywood  I moved the check into the `create_certificates` function now. The form does a try catch so that we see the error in the ` $OUTPUT->error_text` area which I think looks nice.

We unit test that if we call  `create_certificates` directly while the certs are locked, we throw a saml2 exception.